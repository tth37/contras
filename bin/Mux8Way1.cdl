import And4 from "And4.cdl"
import Not from "Not.cdl"
import And from "And.cdl"

@Head
NAME Mux8Way1
INPUT c, b, a
INPUT D0, D1, D2, D3, D4, D5, D6, D7
OUTPUT W, NW
SYMBOL And And1
SYMBOL And4 And41, And42, And43, And44, And45, And46, And47, And48, And49, And410
SYMBOL Not Not1, Not2, Not3, Not4, Not5, Not6, Not7, Not8, Not9, Not10, Not11, Not12

@Body
Not1(a=a)=>(out=na)
Not2(a=b)=>(out=nb)
Not3(a=c)=>(out=nc)
And41(a[0]=nc, a[1]=nb, a[2]=na, a[3]=D0)=>(out=d0)
And42(a[0]=nc, a[1]=nb, a[2]=a, a[3]=D1)=>(out=d1)
And43(a[0]=nc, a[1]=b, a[2]=na, a[3]=D2)=>(out=d2)
And44(a[0]=nc, a[1]=b, a[2]=a, a[3]=D3)=>(out=d3)
And45(a[0]=c, a[1]=nb, a[2]=na, a[3]=D4)=>(out=d4)
And46(a[0]=c, a[1]=nb, a[2]=a, a[3]=D5)=>(out=d5)
And47(a[0]=c, a[1]=b, a[2]=na, a[3]=D6)=>(out=d6)
And48(a[0]=c, a[1]=b, a[2]=a, a[3]=D7)=>(out=d7)
Not5(a=d0)=>(out=nD0)
Not6(a=d1)=>(out=nD1)
Not7(a=d2)=>(out=nD2)
Not8(a=d3)=>(out=nD3)
Not9(a=d4)=>(out=nD4)
Not10(a=d5)=>(out=nD5)
Not11(a=d6)=>(out=nD6)
Not12(a=d7)=>(out=nD7)
And49(a[0]=nD0, a[1]=nD1, a[2]=nD2, a[3]=nD3)=>(out=nD0123)
And410(a[0]=nD4, a[1]=nD5, a[2]=nD6, a[3]=nD7)=>(out=nD4567)
And1(a=nD0123, b=nD4567)=>(out=nresult)
Not4(a=nresult)=>(out=result)

@End
OUTPUT W=result, NW=nresult