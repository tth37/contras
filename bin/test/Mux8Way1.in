@Case
a=01010101010101
b=11111110000000
c=00110011001100
D0=10101010101010
D1=01010101011010
D2=10101010101010
D3=01010101001010
D4=00101000101010
D5=10101101010101
D6=01011111111111
D7=00101110101010
W=?
NW=?
@End