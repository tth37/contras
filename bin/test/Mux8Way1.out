@Case
 a[0] = 01010101010101
 b[0] = 11111110000000
 c[0] = 00110011001100
D0[0] = 10101010101010
D1[0] = 01010101011010
D2[0] = 10101010101010
D3[0] = 01010101001010
D4[0] = 00101000101010
D5[0] = 10101101010101
D6[0] = 01011111111111
D7[0] = 00101110101010
 W[0] = 11001111111110
NW[0] = 00110000000001
